### ISSUE TITLE
how do you use C++ 20 Modules develop the project by VS2019?

### ISSUE THREAD

**Issue by sgf** [fdwr/TextLayoutSampler#3]
I noticed you use the C++ modules.
but Visual Studio so far, no highlighting and Intellisense support.

how do u write so much code without highlighting and Intellisense.

**Comment by fdwr** [comment]
Modules are not *fully* enabled, and I still include headers (see TextLayoutSampler\Common.h where `__cpp_modules` is not defined yet by the compiler). I've been slowly enabling modules the past year as the compiler improves and fixes compiler crashes. Hopefully sometime this year, Visual Studio will finally be mature enough to fully enable modules.

For colored syntax highlighting of .ixx files, go to Tools / Options / Text Editor / File Extension.
Add extension "ixx" as "Microsoft Visual C++"

Regarding Intellisense, I didn't really need it anyway, as I remembered most of the class and function names anyway, since I wrote them all :b (remember the simpler days long ago when we only had Notepad, before complex IDE's?); but Intellisense still works for me because of header file inclusion.
